src/VssCommandSet.cpp

```diff
@@ -28,25 +34,57 @@ std::string VssCommandProcessor::processSet2(WsChannel &channel,
   try {
     requestValidator->validateSet(request);
   } catch (jsoncons::jsonschema::schema_error &e) {
-    logger->Log(LogLevel::ERROR, string(e.what()));
+    std::string msg=std::string(e.what());
+    boost::algorithm::trim(msg);
+    logger->Log(LogLevel::ERROR, msg);
     return JsonResponses::malFormedRequest( requestValidator->tryExtractRequestId(request), "set",
-                                           string("Schema error: ") + e.what());
+                                           string("Schema error: ") + msg);
   } catch (std::exception &e) {
     logger->Log(LogLevel::ERROR, "Unhandled error: " + string(e.what()));
     return JsonResponses::malFormedRequest(
         requestValidator->tryExtractRequestId(request) , "get", string("Unhandled error: ") + e.what());
   }
 
   VSSPath path = VSSPath::fromVSS(request["path"].as_string());
-  bool gen1_compat_mode = path.isGen1Origin();
-
+  
   string requestId = request["requestId"].as_string();
 
   logger->Log(LogLevel::VERBOSE, "Set request with id " + requestId +
                                      " for path: " + path.getVSSPath());
 
+  //unpack any multiset or filters here later as in
+  //list of setPairs=expand(VSSPath, filters)
+
+  std::vector<std::tuple<VSSPath,jsoncons::json>> setPairs;
+  setPairs.push_back(std::make_tuple(path, (jsoncons::json&)request["value"]));
+
+  //Check Access rights first
+  for ( std::tuple<VSSPath,jsoncons::json> setTuple : setPairs) {
+    if (! accessValidator->checkWriteAccess(channel, std::get<0>(setTuple) )) {
+      stringstream msg;
+      msg << "No write  access to " << std::get<0>(setTuple).getVSSPath();
+      return JsonResponses::noAccess(request["requestId"].as<string>(), "set", msg.str());
+    }
+  }
+
+  //Check if exists and is a sensor/actor
+  for ( std::tuple<VSSPath,jsoncons::json> setTuple : setPairs) {
+    if (! database->pathExists(std::get<0>(setTuple) )) {
+      return JsonResponses::pathNotFound(request["requestId"].as<string>(), "set", std::get<0>(setTuple).getVSSPath());
+    }
+    if (! database->pathIsWritable(std::get<0>(setTuple))) {
+      stringstream msg;
+      msg << "Can not set " << std::get<0>(setTuple).getVSSPath() << ". Only sensor or actor leaves can be set.";
+      logger->Log(LogLevel::VERBOSE,msg.str());
+      return JsonResponses::noAccess(request["requestId"].as<string>(), "set", msg.str());
+    } 
+  }
+
+
   try {
-    database->setSignal(channel, path, request["value"], gen1_compat_mode);
+    for ( std::tuple<VSSPath,jsoncons::json> setTuple : setPairs) {
```

Comment:
@SebastianSchildt You looped `setPairs` three times!

Reply:
I was (that is what I told myself) going for readability instead of performance and deeper nesting. I hope the datastructure with the vector of tuples thing access compiles down to just moving a pointer, but I do not know 😕 
Maybe we add an enhancement issue to look into that later?

Reply:
you are right with readability. But in this case, I do not think, that putting everything inside one loop is less readable than using 3 loops. If you like to improve the readability, better refact the method with multiple functions. It is always best practices to have smaller function/method, better than a function with more than 100 lines of code. But If you like, you can create an issue for this, instead of fixing it now.

Reply:
#124